src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -584,11 +589,15 @@ void ProcessOverride (OverrideInformation overrideInformation)
 			// Only track instantiations if override removal is enabled and the type is instantiated.
 			// If it's disabled, all overrides are kept, so there's no instantiation site to blame.
 			if (_context.IsOptimizationEnabled (CodeOptimizations.OverrideRemoval, method) && isInstantiated) {
-				MarkMethod (method, new DependencyInfo (DependencyKind.OverrideOnInstantiatedType, method.DeclaringType), method.DeclaringType);
+				// MarkMethod uses the MessageOrigin passed in here to determine the member which causes the method
+				// to be marked. This is important to avoid spurious warnings as virtual overrides can end up being
+				// marked without being referenced directly from user code. It is important to pass the base method
+				// here so that MarkMethod can see that it was marked because of an override.
+				MarkMethod (method, new DependencyInfo (DependencyKind.OverrideOnInstantiatedType, method.DeclaringType), new MessageOrigin (@base));
```

Comment:
Is this logic still needed to avoid spurious warnings? It looks to me like we don't, since ProcessRequiresUnreferencedCode doesn't have a case for OverrideOnInstantiatedType (or maybe it's missing a case?)

More generally (not directly related to your change) - I'm not sure what the function of sourceLocationMember is here - I think we should just pass null. If I'm not missing something, it looks like:
- At this point, method and @base are MethodDefinitions
- MarkMethod will only use sourceLocationMember for instantiated generic types or methods (the latter in GetOriginalMethod)

I think the sourceLocationMember is only used for instantiated generics, but can't have instantiated generics here, so we shouldn't use it. Does that sound right to you? @vitek-karas I think you introduced these - what do you think?

Reply:
You are right it's no longer needed; when calling MarkMethod from here we will never emit a warning.
> More generally (not directly related to your change) - I'm not sure what the function of sourceLocationMember is here - I think we should just pass null.

Yes, since there's no way we will emit a warning in MarkMethod at this point, we can pass null here. Other than helping us report the warning, sourceLocationMember becomes relevant in the context of MarkMethod when dealing with generics, as you noted (we could end here:

https://github.com/mono/linker/blob/ba990ce8336dd278b31007aac30ac85f120d3683/src/linker/Linker.Steps/MarkStep.cs#L2267

 I'll change the origin here to null, unless @vitek-karas disagrees / there a reason to pass @base/method.DeclaringType here.